docs/concepts/models/sql_models.md

```diff
@@ -40,20 +43,70 @@ SELECT
+  executed concurrently with other statements, such as creating physical tables.
+*/
+UNCACHE TABLE countries;
 ```
 
-### Model DDL
-The `MODEL` DDL is used to specify metadata about the model such as name, [kind](./model_kinds.md), owner, cron, and others. The Model statement should be the first statement in your model SQL file.
+### `MODEL` DDL
+The `MODEL` DDL is used to specify metadata about the model such as its name, [kind](./model_kinds.md), owner, cron, and others. This should be the first statement in your SQL-based model's file.
+
+Refer to `MODEL` [properties](./overview.md#properties) for the full list of allowed properties.
+
+### Optional pre/post-statements
+Optional pre/post-statements can help prepare the model's query and execute "clean-up" actions after it has successfully executed, respectively. For example, you might create temporary views, set permissions, or evict previously cached tables from memory.
+
+However, be careful not to run any statement that could conflict with the execution of another statement when ran concurrently, such as creating a physical table.
+
+### The model query
+The model must contain a standalone query, which can be a single `SELECT` expression, or multiple `SELECT` expressions combined with the `UNION`, `INTERSECT`, or `EXCEPT` operators. The result of this query will be used to populate the model's table or view.
+
+## Python-based definition
 
-Refer to `model` [properties](./overview.md#properties) for the full list of allowed properties.
+The Python-based definition of SQL models consists of a single python function, decorated with SQLMesh's `@model` [decorator](https://wiki.python.org/moin/PythonDecorators). The decorator is required to have the `is_sql` keyword argument set to `True` to distinguish it from [Python models](./python_models.md) that return DataFrame instances.
 
-### Optional statements
-Optional SQL statements can help you prepare the model's `SELECT` statement. For example, you might create temporary tables or set permissions.
+This function's return value serves as the model's query, and it must be either a SQL string, or a [SQLGlot expression](https://github.com/tobymao/sqlglot/blob/main/sqlglot/expressions.py). The `@model` decorator is used to define the model's [metadata](#MODEL-DDL) and, optionally, its pre/post-statements that are also in the form of SQL strings or SQLGlot expressions.
 
-However, be careful not to run any command that could conflict with the execution of the model's query, such as creating a physical table.
+Defining a model like this can be beneficial in cases where its query is too complex to express cleanly in SQL, for example due to having many dynamic components that would require heavy use of [macros](../macros/overview/). Since Python-based models generate SQL, they also support column-level [lineage](../glossary/#lineage), [automatic change categorization](../glossary/#automatic-data-rebasing), etc.
```

Comment:
"Defining a model like this" --> "Defining a SQL model with Python"

"they also support" --> "they support the same features as regular SQL models, such as"